python/pylibraft/pylibraft/test/test_config.py

```diff
@@ -0,0 +1,54 @@
+# Copyright (c) 2022, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import cupy
+import pytest
+
+import pylibraft.config
+from pylibraft.common import auto_convert_output, device_ndarray
+
+pytest.importorskip("cupy")
+
+
+@auto_convert_output
+def gen_cai(m, n, t=None):
+    if t is None:
+        return device_ndarray.empty((m, n))
+    elif t == tuple:
+        return device_ndarray.empty((m, n)), device_ndarray.empty((m, n))
+    elif t == list:
+        return [device_ndarray.empty((m, n)), device_ndarray.empty((m, n))]
+
+
+@pytest.mark.parametrize(
+    "out_type",
+    [["cupy", cupy.ndarray], ["raft", pylibraft.common.device_ndarray]],
```

Comment:
Might be worth testing out the `callable` path here too?  I think we could test out setting the output as numpy pretty easy:

```suggestion
    [["cupy", cupy.ndarray], ["raft", pylibraft.common.device_ndarray], [lambda arr: arr.copy_to_host(), np.ndarray]],
```

(note that I haven't tested this out at all - and I'd be wary of applying the suggestion directly =)

Reply:
Looks like this works!